cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/sidecar/ArtifactLocalizerInternalAuthenticator.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.worker.sidecar;
+
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.internal.remote.InternalAuthenticator;
+import io.cdap.cdap.proto.security.Credential;
+import io.cdap.cdap.proto.security.GcpMetadataTaskContext;
+import java.util.function.BiConsumer;
+import javax.annotation.Nullable;
+
+/**
+ * A class which sets internal authenticated headers for the remote client using a
+ * {@link GcpMetadataTaskContext} as the source.
+ */
+public class ArtifactLocalizerInternalAuthenticator implements InternalAuthenticator {
```

Comment:
nit: Prefer to rename this to `GcpWorkloadIdentityInternalAuthenticator` to reflect what it does, wdyt?

Reply:
ack